service/src/java/org/apache/hive/service/server/HiveServer2.java

```diff
@@ -534,6 +455,101 @@ public synchronized void init(HiveConf hiveConf) {
     // Extra time for releasing the resources if timeout sets to 0
     ShutdownHookManager.addGracefulShutDownHook(() -> graceful_stop(),  timeout == 0 ? 30 : timeout);
   }
+  
+  private void addHAContextAttributes(HttpServer.Builder builder, HiveConf hiveConf) {
+    builder.setContextAttribute("hs2.isLeader", isLeader);
+    builder.setContextAttribute("hs2.failover.callback", new FailoverHandlerCallback(hs2HARegistry));
+    builder.setContextAttribute("hiveconf", hiveConf);
+  }
+  
+  private HttpServer.Builder initBuilder(String webHost, int port, String name, String contextPath, HiveConf hiveConf) throws IOException {
+    HttpServer.Builder builder = new HttpServer.Builder(name);
+    builder.setPort(port).setConf(hiveConf);
```

Comment:
can we separate the `port` and `hiveConf` setup?

Reply:
That existed in the original code which I refactored into another method.
Done.